.ci/jobDSL_generator.groovy

```diff
@@ -0,0 +1,42 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+pipeline {
+    agent {label 'master'}
+    stages {
+        stage('Generate Jobs') {
+            steps {
+                checkout([$class: 'GitSCM',
+                  branches: [[name: '*/jobDSL']],
+                  doGenerateSubmoduleConfigurations: false,
+                  extensions: [],
+                  submoduleCfg: [],
+                  userRemoteConfigs: [[
+                    credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7-UserAndToken',
+                    url: 'http://github.com/kuisathaverat/apm-pipeline-library.git'
+                ]]])
```

Comment:
What's the reason for this checkout instead the traditional one?
should the url be changed?

Reply:
if we will have more than one repo, so one checkout per repo, and one jobDsl step to process the jobs.

Reply:
>should the url be changed?

That URL is only for testing locally